privacy-protections/fingerprinting/helpers/tests.js

```diff
@@ -0,0 +1,1292 @@
+        category: 'headers',
+        getValue: () => headers.then(res => res.headers['accept-encoding'])
+    },
+    {
+        id: 'headers - accept-language',
+        category: 'headers',
+        getValue: () => headers.then(res => res.headers['accept-language'])
+    },
+    {
+        id: 'headers - dnt',
+        category: 'headers',
+        getValue: () => headers.then(res => res.headers['dnt'])
+    },
+    {
+        id: 'headers - user-agent',
+        category: 'headers',
+        getValue: () => headers.then(res => res.headers['user-agent'])
+    },
+    {
+        id: 'headers - other',
+        category: 'headers',
+        getValue: () => headers.then(res => {
+            const exclude = [
+                'accept', 'accept-encoding', 'accept-language', 'dnt', 'user-agent', // already covered above
+                'referer' // not useful
+            ];
+            const other = {};
+
+            Object.keys(res.headers).forEach(k => {
+                if (!exclude.includes(k)) {
+                    other[k] = res.headers[k];
+                }
+            })
+
+            return other;
+        })
+    },
+
+    // navigator
+    {
+        id: 'navigator.userAgent',
+        category: 'navigator',
+        getValue: () => navigator.userAgent
+    },
+    {
+        id: 'navigator.deviceMemory',
+        category: 'navigator',
+        getValue: () => navigator.deviceMemory
+    },
+    {
+        id: 'navigator.hardwareConcurrency',
+        category: 'navigator',
+        getValue: () => navigator.hardwareConcurrency
+    },
+    {
+        id: 'navigator.product',
+        category: 'navigator',
+        getValue: () => navigator.product
+    },
+    {
+        id: 'navigator.productSub',
+        category: 'navigator',
+        getValue: () => navigator.productSub
+    },
+    {
+        id: 'navigator.appName',
+        category: 'navigator',
+        getValue: () => navigator.appName
+    },
+    {
+        id: 'navigator.appVersion',
+        category: 'navigator',
+        getValue: () => navigator.appVersion
+    },
+    {
+        id: 'navigator.appCodeName',
+        category: 'navigator',
+        getValue: () => navigator.appCodeName
+    },
+    {
+        id: 'navigator.language',
+        category: 'navigator',
+        getValue: () => navigator.language
+    },
+    {
+        id: 'navigator.languages',
+        category: 'navigator',
+        getValue: () => navigator.languages
+    },
+    {
+        id: 'navigator.platform',
+        category: 'navigator',
+        getValue: () => navigator.platform
+    },
+    {
+        id: 'navigator.vendor',
+        category: 'navigator',
+        getValue: () => navigator.vendor
+    },
+    {
+        id: 'navigator.vendorSub',
+        category: 'navigator',
+        getValue: () => navigator.vendorSub
+    },
+    {
+        id: 'navigator.mimeTypes',
+        category: 'navigator',
+        getValue: () => {
+            const results = {};
```

Comment:
in many cases I, somehow forcefully, create objects rather than arrays (even if arrays would make more sense). The reason is diffing - if one element in the middle of the array is missing the diff will be huge because all other items are shifted, for objects this is not a problem.